src/pushsource/_impl/backend/errata_source/errata_client.py

```diff
@@ -1,16 +1,43 @@
-import logging
+from functools import wraps, partial
+import gssapi
+import os
+import re
+import requests
+import requests_gssapi
+import subprocess
+import tempfile
 import threading
-from functools import partial
-
+from urllib.parse import urljoin
 import xmlrpc.client as xmlrpc_client
+
+import logging
+
 from more_executors import Executors
 from more_executors.futures import f_zip, f_map
 
 from ...compat_attr import attr
 
+
 LOG = logging.getLogger("pushsource.errata_client")
 
 
+USE_HTTP_CLIENT = os.environ.get("PUSHSOURCE_USE_ERRATA_HTTP_API") == "1"
+
+
+def get_errata_client(
+    threads,
+    url,
+    keytab_path=None,
+    principal=None,
+    force_http=USE_HTTP_CLIENT,
+    **retry_kwargs,
+):
+    if force_http:
```

Comment:
Should it be also trying to use the HTTP client if keytab/principal were provided?

If it doesn't, I think that would already break the `product_name` support added recently.  Because the implemented behavior was, "you get product_name if you pass keytab/principal", but with this logic as you've got it here it'd instead be "you get product_name only if PUSHSOURCE_USE_ERRATA_HTTP_API is set to 1".

Reply:
I had intended for the `PUSHSOURCE_USE_ERRATA_HTTP_API` to replace product_name values. Given what you mentioned before about the variability of environments, I liked the idea of having some sort of "handbrake". Maybe it can be flipped around, so it'll use HTTP by default if keytab/principal are provided, but we have an envvar that can force it to use xmlrpc

Reply:
I don't understand what "replace product_name values" means there.

Let me put it this way.  This is what I expect the state of the world to be leading up to deployment of this change:

1. Some environments using this library are providing keytab/principal, and are expecting to see non-null `product_name`.
2. Some environments using this library are not providing keytab/principal, and don't care about `product_name`.

We must not break either of those cases.

i.e. when upgrading to the version of pushsource with this change, in case (1), `product_name` must not suddenly disappear, and in case (2), keytab/principal must not suddenly become mandatory.

The most obvious way to achieve this seems to be making it default to the HTTP client (which supports `product_name`) if keytab/principal is provided, and default to the XML-RPC client if keytab/principal is not provided.

An env var on top of this to force one or the other is fine but I'd see that as being only for emergency usage, if something isn't working how we expect it to after deployment.